helpers.md

```diff
@@ -585,6 +585,42 @@ async function main() {
 main();
 ```
 
+#### Use `afterCompletion` for multi-agent patterns
+
+The `afterCompletion` callback allows for some powerful multi-agent patterns. By passing runner.messages to another LLM chat within `afterCompletion`, you can easily have another model analyze the conversation and do things like conditionally inject web research or other relevant data or guidance to help the first model overcome problems.
+
+```ts
+import OpenAI from 'openai';
+
+const client = new OpenAI();
+
+async function main() {
+  let shouldInjectMessage = false // You can do any kind of conditional logic you want
+  const runner = client.chat.completions
+    .runTools({
+      model: 'gpt-3.5-turbo',
+      messages: [{ role: 'user', content: "How's the weather this week in Los Angeles?" }],
+      tools: [
+        // Whole bunch of tools...perhaps so many that we need to offload some cognitive overhead to another chat via afterCompletion...
+      ],
+    },
+    {
+      afterCompletion: async () => {
+        if (!shouldInjectMessage) {
+          runner._addMessage({
+            role: 'system',
```

Comment:
I know this is just an example, but I think we'd want this to be an assistant message also

Reply:
Actually, I'm a bit unclear on this: I've been using `system` for these kinds of messages where I "inject" important context or attempt to give the model a strong nudge to go in a different direction or update its instructions partway through a chat, and that seems to work well for me. (I also display all `assistant` messages in my FE by default and wouldn't want to display this message, but I could find a way around that if needed.) Could I potentially get better results in such cases by using the `assistant` role and changing up how I word the message content accordingly? I'm consistently getting the model to [respond intelligently to inline web research](https://www.loom.com/share/3b0766fa0e3340c1b51c9dfddc8f6fcc?t=892&sid=e71628a0-1397-4cae-9e8b-4fec7abbb642) using my current approach, but I'm curious and honestly I've probably neglected the `assistant` role a bit when it comes to manually inserted messages!